src/allmydata/test/web/test_root.py

```diff
@@ -36,65 +36,36 @@ class RenderSlashUri(unittest.TestCase):
     """
 
     def setUp(self):
-        self.request = DummyRequest(b"/uri")
-        self.request.fields = {}
-
-        def prepathURL():
-            return b"http://127.0.0.1.99999/" + b"/".join(self.request.prepath)
-
-        self.request.prePathURL = prepathURL
         self.client = Mock()
         self.res = URIHandler(self.client)
 
     def test_valid(self):
         """
         A valid capbility does not result in error
         """
-        self.request.args[b"uri"] = [(
+        query_args = {b"uri": [
             b"URI:CHK:nt2xxmrccp7sursd6yh2thhcky:"
             b"mukesarwdjxiyqsjinbfiiro6q7kgmmekocxfjcngh23oxwyxtzq:2:5:5874882"
-        )]
-        self.res.render_GET(self.request)
-
-    def test_invalid(self):
-        """
-        A (trivially) invalid capbility is an error
-        """
-        self.request.args[b"uri"] = [b"not a capability"]
-        response_body = self.res.render_GET(self.request)
-
-        soup = BeautifulSoup(response_body, 'html5lib')
-
-        assert_soup_has_tag_with_content(
-            self, soup, "title", "400 - Error",
```

Comment:
This error used to get rendered as HTML but I think that was a mistake.  There is explicit support code in the Nevow code we have for handling `WebError` that this code exercises and rendering a text-only response describing it.  The first version of `@render_exception` unintentionally preempted that code and turned the response into html.